File: app/src/main/java/com/dxm/aimodel/base/HeaderBar.java
package com.dxm.aimodel.base;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.dxm.aimodel.R;

/**
 * Author: Meng
 * Date: 2024/03/30
 * Modify: 2024/03/30
 * Desc:
 */
public class HeaderBar extends FrameLayout {

    public HeaderBar(@NonNull Context context) {
        this(context, null);
    }

    public HeaderBar(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public HeaderBar(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    public HeaderBar(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initView(context, attrs);
    }

    public void initView(Context context, AttributeSet attrs) {
        FrameLayout layout = (FrameLayout) LayoutInflater.from(context).inflate(R.layout.bar_header,this,false);
        addView(layout);
        getParams(context, attrs);
    }

    private void getParams(Context context, AttributeSet attrs){
        //获取参数
        TypedArray a = context.obtainStyledAttributes(attrs,R.styleable.HeaderBar);

        boolean isBack = a.getBoolean(R.styleable.HeaderBar_back, true);
        String title = a.getString(R.styleable.HeaderBar_title);
        String rightText = a.getString(R.styleable.HeaderBar_rightText);
        int rightIcon = a.getResourceId(R.styleable.HeaderBar_rightIocn, R.drawable.ic_launcher_background);

        ImageView backBtn = findViewById(R.id.header_back_btn);
        TextView titleView = findViewById(R.id.header_title);
        TextView rightView = findViewById(R.id.header_right_text);
        ImageView rightBtn = findViewById(R.id.header_right_btn);

        if(!isBack) {
            backBtn.setVisibility(GONE);
        }
        if(title != null) {
            titleView.setText(title);
        }

        if(rightText != null) {
            rightView.setText(rightText);
            rightView.setVisibility(VISIBLE);
        }

        if(rightIcon != R.drawable.ic_launcher_background) {
            rightBtn.setImageResource(rightIcon);
            rightBtn.setVisibility(VISIBLE);
        }
//        int num = a.getInt(R.styleable.HeaderBar_num, 0);
//        int num2 = a.getFloat(R.styleable.HeaderBar_num, 0);
//        float num3 = a.getDimension(R.styleable.HeaderBar_num, 0);
//        int color = a.getColor(R.styleable.HeaderBar_color, "#000000");

        a.recycle();
    }


}

File: app/src/main/java/com/ql/health/custom/widget/HeaderBar.kt
package com.ql.health.custom.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import com.ql.health.R


/**
 * Author: Meng
 * Date: 2024/11/21
 * Modify: 2024/11/21
 * Desc:
 */
class HeaderBar @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0,
    defStyleRes: Int = 0
) :
    FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
    init {
        initView(context, attrs)
    }

    fun initView(context: Context, attrs: AttributeSet?) {
        val layout =
            LayoutInflater.from(context).inflate(R.layout.bar_header, this, false) as FrameLayout
        addView(layout)
        getParams(context, attrs)
    }

    private fun getParams(context: Context, attrs: AttributeSet?) {
        //获取参数
        val a = context.obtainStyledAttributes(attrs, R.styleable.HeaderBar)

        val isBack = a.getBoolean(R.styleable.HeaderBar_back, true)
        val title = a.getString(R.styleable.HeaderBar_title)
        val rightText = a.getString(R.styleable.HeaderBar_rightText)
        val rightIcon =
            a.getResourceId(R.styleable.HeaderBar_rightIocn, R.drawable.ic_launcher_background)

        val backBtn = findViewById<ImageView>(R.id.header_back_btn)
        val titleView = findViewById<TextView>(R.id.header_title)
        val rightView = findViewById<TextView>(R.id.header_right_text)
        val rightBtn = findViewById<ImageView>(R.id.header_right_btn)

        if (!isBack) {
            backBtn.visibility = GONE
        }
        if (title != null) {
            titleView.text = title
        }

        if (rightText != null) {
            rightView.text = rightText
            rightView.visibility = VISIBLE
        }

        if (rightIcon != R.drawable.ic_launcher_background) {
            rightBtn.setImageResource(rightIcon)
            rightBtn.visibility = VISIBLE
        }

        //        int num = a.getInt(R.styleable.HeaderBar_num, 0);
//        int num2 = a.getFloat(R.styleable.HeaderBar_num, 0);
//        float num3 = a.getDimension(R.styleable.HeaderBar_num, 0);
//        int color = a.getColor(R.styleable.HeaderBar_color, "#000000");
        a.recycle()
    }
}